apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,745 @@
+    * @param _voter Address of the voter
+    * @return Address of the caster of the voter's vote
+    */
+    function getVoteCaster(uint256 _voteId, address _voter) public view voteExists(_voteId) returns (address) {
+        return _voteCaster(votes[_voteId], _voter);
+    }
+
+    /**
+    * @notice Tells whether `_representative` is `_voter`'s representative or not
+    * @return True if the given representative was allowed by a certain voter, false otherwise
+    */
+    function isRepresentativeOf(address _voter, address _representative) public view isInitialized returns (bool) {
+        return _isRepresentativeOf(_voter, _representative);
+    }
+
+    /**
+    * @notice Tells whether vote #`_voteId` is within the overrule period for delegated votes or not
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote(),` which requires initialization
+    * @return True if the requested vote is within the overrule window for delegated votes, false otherwise
+    */
+    function withinOverruleWindow(uint256 _voteId) public view voteExists(_voteId) returns (bool) {
+        Vote storage vote_ = votes[_voteId];
+        return _isVoteOpen(vote_) && _withinOverruleWindow(vote_);
+    }
+
+    // Internal fns
+
+    /**
+    * @dev Internal function to create a new vote
+    * @return voteId id for newly created vote
+    */
+    function _newVote(bytes _executionScript, string _metadata, bool _castVote) internal returns (uint256 voteId) {
+        uint64 snapshotBlock = getBlockNumber64() - 1; // avoid double voting in this very block
+        uint256 votingPower = token.totalSupplyAt(snapshotBlock);
+        require(votingPower > 0, ERROR_NO_VOTING_POWER);
+
+        voteId = votesLength++;
+
+        Vote storage vote_ = votes[voteId];
+        vote_.startDate = getTimestamp64();
+        vote_.snapshotBlock = snapshotBlock;
+        vote_.overruleWindow = overruleWindow;
+        vote_.supportRequiredPct = supportRequiredPct;
+        vote_.minAcceptQuorumPct = minAcceptQuorumPct;
+        vote_.votingPower = votingPower;
+        vote_.executionScript = _executionScript;
+        vote_.actionId = _newAgreementAction(voteId, msg.sender, bytes(_metadata));
+        vote_.disputableStatus = DisputableStatus.Active;
+
+        emit StartVote(voteId, msg.sender, _metadata);
+
+        if (_castVote && _canVote(vote_, msg.sender)) {
+            _vote(voteId, true, msg.sender);
+        }
+    }
+
+    /**
+    * @dev Internal function to cast a vote. It assumes the queried vote exists.
+    */
+    function _vote(uint256 _voteId, bool _supports, address _voter) internal {
+        // This could re-enter, though we can assume the governance token is not malicious
+        Vote storage vote_ = votes[_voteId];
+        _castVote(vote_, _voteId, _supports, _voter);
+        _overwriteCasterIfNecessary(vote_, _voter);
+
+        if (_canExecute(vote_)) {
```

Comment:
I think we were discussing removing early execution; did we settle on removing it from this version?

Reply:
I think we did, but happy to discuss if you think it's a bad idea or if there is another alternative around it

Reply:
Nope, let's lift it as an assumption / something to confirm.

With disputables, any concept of early execution feels harmful.